### PR TITLE
allow enabling kubelet serving certificate rotation via flag

### PR DESCRIPTION
**Pull request by liggitt**
allows the server rotation feature to be exercised via flags as well as config file

```release-note
NONE
```

### CODE REVIEW

**Comment by mikedanese** [comment]
but why???

cc @awly

**Comment by liggitt** [comment]
to let it be exercised by existing components that use flags (e.g. CI)

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -489,6 +489,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 		"If --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key "+
 		"are generated for the public address and saved to the directory passed to --cert-dir.")
 	fs.StringVar(&c.TLSPrivateKeyFile, "tls-private-key-file", c.TLSPrivateKeyFile, "File containing x509 private key matching --tls-cert-file.")
+	fs.BoolVar(&c.ServerTLSBootstrap, "rotate-server-certificates", c.ServerTLSBootstrap, "<Warning: alpha feature> Auto request and rotate the kubelet serving certificates by requesting new certificates from the kube-apiserver when the certificate expiration approaches. Requires the RotateKubeletServerCertificate feature gate to be enabled.")
```
Comment (mikedanese) [inline cmd/kubelet/app/options/options.go]: Is this worth setting as deprecated?

**Comment by mikedanese** [comment]
lgtm
/approve

cc @mtaufen

File: cmd/kubelet/app/options/options.go (hunk 1 above)
Comment (liggitt) [reply cmd/kubelet/app/options/options.go]: the `defer func()` at the top of the function auto-marks all the flags bound to config struct fields as deprecated with an explanation

**Comment by mtaufen** [comment]
The beta file-based config API already provides a stronger API stability guarantee than the GA flags (see https://github.com/kubernetes/kubernetes/pull/65191#issuecomment-398181931).
Given this, I'm not convinced we need to add flags for every parameter in the config. This seems like a step backward vs. our goal of migration to the file.

**Comment by liggitt** [comment]
> Given this, I'm not convinced we need to add flags for every parameter in the config. This seems like a step backward vs. our goal of migration to the file.

this is an existing feature that was very usable via flags alone in 1.10 (especially when paired with out-of-band approval flows), and got promoted to beta in 1.11 in the kubelet. maintaining parity between flags and config for features like that makes sense to me.

**Comment by awly** [comment]
This SGTM, I was really surprised to see we don't auto-generate flags from config.

**Comment by liggitt** [comment]
> This SGTM, I was really surprised to see we don't auto-generate flags from config.

I agree with @mtaufen that we don't need/want to maintain parity between flags and config for all new features forever (especially features that require structured config information are very difficult to express as flags).

I would push to keep parity for existing or in-progress features, and on a case-by-case basis for simple features, even as we continue positioning the flags as a whole as deprecated and mature the kubelet config file.

@mtaufen/@mikedanese want to tag this PR?

**Pull request merged**